server/src/weaverbird/backends/mongo_translator/steps/date_extract.py

```diff
@@ -0,0 +1,320 @@
+from copy import deepcopy
+from typing import List
+
+from weaverbird.backends.mongo_translator.steps.types import MongoStep
+from weaverbird.pipeline.steps import DateExtractStep
+
+
+def _truncate_date_to_day(expr: dict) -> MongoStep:
```

Comment:
On js : 
```javascript
export function truncateDateToDay(dateExpr: string | object): object {
```

i think, the translation should be : 
```suggestion
def _truncate_date_to_day(expr: Union[str, dict]) -> MongoStep:
```

What do you think ?

Reply:
That's true, but I believe the TS version is out-of-date (or maybe I missed something). In this file, it is only used for dicts, so I wanted the type to be as specific as possible. Maybe we can change it if we also use it for strings at some point ? Or I can change it now, I don't have a strong opinion on this one